astro_2.0/features/Notifications/helpers.tsx

```diff
@@ -369,3 +363,85 @@ export function mapDelayToTime(delay: string): number {
     }
   }
 }
+
+const mapNotificationTypeToMessage = (
+  type: NotifiedActionType,
+  t: TFunction
+) => {
+  switch (type) {
+    case NotifiedActionType.AddMemberToRole: {
+      return t('addMemberToGroup');
+    }
+    case NotifiedActionType.RemoveMemberFromRole: {
+      return t('removeMemberFromGroup');
+    }
+    case NotifiedActionType.AddBounty: {
+      return t('createBounty');
+    }
+    case NotifiedActionType.BountyDone: {
+      return t('bountyDone');
+    }
+    case NotifiedActionType.ChangeConfig: {
+      return t('changeConfig');
+    }
+    case NotifiedActionType.ChangePolicy: {
+      return t('changePolicy');
+    }
+    case NotifiedActionType.Transfer: {
+      return t('transfer');
+    }
+    case NotifiedActionType.FunctionCall: {
+      return t('functionCall');
+    }
+    case NotifiedActionType.Vote: {
+      return t('poll');
+    }
+    default: {
+      return '';
+    }
+  }
+};
+
+export function generateProposalNotificationText(
+  accountId: string,
+  proposerId: string,
+  status: NotificationStatus,
+  notificationType: NotifiedActionType,
+  dao: DAO,
+  t: TFunction
+): string {
+  const type = mapNotificationTypeToMessage(notificationType, t);
+
+  switch (status) {
+    case NotificationStatus.Created: {
+      const actioner =
+        accountId === proposerId ? 'You' : extractPrefix(proposerId);
+
+      return `${actioner} ${t('submittedNewProposal', {
+        type,
+        dao: dao.displayName || extractPrefix(dao.id),
+      })}`;
+    }
+    case NotificationStatus.Rejected: {
+      const actioner =
+        accountId === proposerId ? 'Your' : extractPrefix(proposerId);
+
+      return `${actioner}'s ${type} ${t('proposalWasRejected', {
+        dao: dao.displayName || extractPrefix(dao.id),
+      })}`;
+    }
+    case NotificationStatus.Approved: {
+      const actioner =
+        accountId === proposerId ? 'Your' : extractPrefix(proposerId);
```

Comment:
I see that some strings are localised and some are not. Not sure if you've left some intentionally.

Reply:
fixed